pkg/controller/nodeipam/ipam/cloud_cidr_allocator_test.go

```diff
@@ -398,14 +424,563 @@ func TestUpdateCIDRAllocation(t *testing.T) {
 			},
 			expectedUpdate: true,
 		},
+		{
```

Comment:
/hold
I'm holding because you may want to review these test cases and  check the expected results since I'm not familiar with the logic, feel free to unhold just typing the command `/hold cancel` on one comment if you feel is ok

/lgtm
/appprove